src/OpenTelemetry.AutoInstrumentation/Diagnostics/SdkSelfDiagnosticsEventListener.cs

```diff
@@ -76,35 +76,38 @@ protected override void OnEventSourceCreated(EventSource eventSource)
     /// <param name="eventData">Data of the EventSource event.</param>
     protected override void OnEventWritten(EventWrittenEventArgs eventData)
     {
-        object[] payloadArray;
-        if (eventData.Payload != null)
+        if (eventData.EventSource.Name.StartsWith(EventSourceNamePrefix, StringComparison.OrdinalIgnoreCase))
```

Comment:
maybe revert if to return early and reduce nesting and a huge scope?

Reply:
changed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3616/commits/eccc7e5b584a5c7d0982bc6fd4410305386fa364